pkg/knowledge_base2/resource_template.go

```diff
@@ -81,7 +81,7 @@ type (
 		// GetDefaultValue returns the default value for the property, pertaining to the specific data being passed in for execution
 		GetDefaultValue(ctx DynamicContext, data DynamicValueData) (any, error)
 		// Validate ensures the value is valid for the property and returns an error if it is not
-		Validate(value any, properties construct.Properties) error
+		Validate(resource *construct.Resource, value any) error
```

Comment:
Why does it need `value`, can't it get it from the resource from its path? What does it need the resource (was the Properties) for? As far as I could tell from this diff, it's unused.

Reply:
if you want to validate a value before its being set you would need to pass in the value, so we can either make this only check the current property, but then we lose some prechecking ability

Reply:
the properties (and now resource) is going to be needed for validity checks that are more complex (if field a is blah, this must be TCP, etc)